src/rules/no-interpolation-inline-snapshot.ts

```diff
@@ -0,0 +1,51 @@
+import { AST_NODE_TYPES } from '@typescript-eslint/experimental-utils';
+import { createRule, isExpectCall, parseExpectCall } from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Disallow string interpolation inside inline snapshots.',
+      recommended: false,
+    },
+    messages: {
+      noInterpolation:
+        'Do not use string interpolation inside of inline snapshots',
+    },
+    schema: [],
+    type: 'suggestion',
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      CallExpression(node) {
+        if (!isExpectCall(node)) {
+          return;
+        }
+
+        const { matcher } = parseExpectCall(node);
+
+        if (
+          matcher?.name !== 'toMatchInlineSnapshot' ||
+          matcher?.arguments === null
```

Comment:
I think you might as well just use optional chaining on the `forEach` below, since `matcher?.name !==` will also check that `matcher` is defined

Reply:
I am not sure I understand what you mean here.

Reply:
Optional chaining is cool, but if you use it multiple times in a row on the same object, that's usually a smell that you should just check for existence.

When using optional chaining in a condition, it has the same guarding as doing "obj && obj.prop <condition>", meaning it does a "is defined" check.

For example this:

```
if(node.parent && node.parent.type === AST_NODE_TYPES.AssignmentExpression) {
  const { right } = node.parent;

  ...
}
```

Can be simplified to just:

```
if(node.parent?.type === AST_NODE_TYPES.AssignmentExpression) {
  const { right } = node.parent;

  ...
}
```

----------

So originally what I was saying when your code was just `matcher?.name === 'value'` was that you could drop the `matcher.arguments === null` in favor of just later doing `matcher.arguments?.forEach`.

However, now that the code has changed to include multiple compares to name, for which I'd recommend using `include`, and so in turn just do a `matcher &&` as `matcher?.name` returns type `string | undefined` which is not accepted by `string[].includes` (as that expects just `string`).

----------

tl;dr

* I would replace the multiple `matcher?.name ===`s with just `matcher && ['toMatchInlineSnapshot', 'toThrowErrorMatchingInlineSnapshot'].includes(matcher.name)`
* I would replace `matcher.arguments === null` with `matcher.arguments?.forEach`

Reply:
changed in https://github.com/jest-community/eslint-plugin-jest/pull/553/commits/96eaf9b56e404c1c6f835792232067fdc7af9151